robottelo/__init__.py

```diff
@@ -1,9 +1,15 @@
 """This module contains helper code used by :mod:`tests.foreman` module.
 
-This module is subservient to :mod:`tests.foreman`, and exists soley for the
+This module is subservient to :mod:`tests.foreman`, and exists solely for the
 sake of helping that module get its work done. For example,
 :mod:`tests.foreman.cli` relies upon :mod:`robottelo.cli`.  More generally:
 code in :mod:`tests` calls code in :mod:`robottelo`, but not the other way
 around.
 
 """
+from robottelo.config import settings
+
+
+# Setting the configuration for all the robottelo helpers if its not set.
+if not settings.configured:
+    settings.configure()
```

Comment:
You can't do that, since `robottelo.properties` is empty when travis CI tests are run.

See https://github.com/SatelliteQE/robottelo/pull/8003 , "Remaining work", point 5.

Reply:
Yes, but robottelo helpers in `robottelo` directory should be called only when the tests are triggered (if I am not wrong, by looking at modules in robottelo directory), right ? At least before triggering tests robottelo.properties should be being populated.

Else We could also move this to `tests.__init__.py`, whats your thought on that ?

Reply:
`robottelo/__init__.py` is executed whenever `robottelo` appears in import statement. Including things like `from robottelo.constants import VALID_GCE_ZONES`.

My thought is that separate module (that will be ever imported after settings), or new helper function, are the right solution.
Yes, they require updating all callers, but that is very static change. I would see if Pycharm refactoring modules could help here.

If you need this merged yesterday and can't refactor callers, then having `settings.configure()` function adding these constants to `constants` module would be acceptable.